bokeh/models/annotations/html/html_annotation.py

```diff
@@ -0,0 +1,57 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2021, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+'''
+
+'''
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Bokeh imports
+from ....core.has_props import abstract
+from ..annotation import Annotation
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "HTMLAnnotation",
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+@abstract
+class HTMLAnnotation(Annotation):
+    ''' Base class for HTML-based annotations.
+
+    All annotations that inherit from this base class can be attached to
+    a canvas, but are not rendered to it, thus they won't appear in saved
+    plots (only export tools can preserve HTML annotations).
```

Comment:
cc @tcmetzger this is probably an important point to emphasize about these new "HTML annotations". Not sure if making this an admonition would suffice or if we should make space in the users guide too.

Reply:
I guess it would also be good to provide some minimal guidance about why/when a user would want to use HTML versions of annotations.

Reply:
Also "export tools" is too broad, as it's only `export_png()` (which really should have been called `capture_png()`, as in screen capture).

Reply:
Agree on narrowing to specify `export_png`